### DISCUSSION TITLE
Birdnet-pi - stopped recording since update

### DISCUSSION THREAD

**Discussion by PetrusMax** [mcguirepr89/BirdNET-Pi#500] [General]
I had a birdnet running very well for several days until an update this morning.  Then it has stopped recording and sharing any information to birdweather.  Anyone else have the same issue ?

**Comment by ehpersonal38** [comment]
Can you provide the output of your "View Logs"?

**Reply by PetrusMax** [reply]
sorry but view logs only shows "connection closed"

**Reply by ehpersonal38** [reply]
Hmm, I'm not exactly sure what that log message means. Does everything return to normal when you restart the services? (Tools -> Services -> Restart core services)

**Comment by PetrusMax** [comment]
Hi. 

I have tried. Restarting core services, rebooting but nothing helps.  

Issues started following the system update i did earlier today


sent from my ithing

**Reply by ehpersonal38** [reply]
Okay can you please provide the output of the following commands? 
 `bash ~/BirdNET-Pi/scripts/restart_services.sh` 
(it will take up to a minute).

This one may be helpful too:
`journalctl --no-hostname --no-pager -n100 -u birdnet_analysis -u birdnet_server -u birdnet_recording | sed "s|$HOME||g"`

**Reply by PetrusMax** [reply]
i sent the output by email.  If I paste here all test is marked with a strikethrough. I dont know why.

**Comment by PetrusMax** [comment]
I have noticed that I can get BirdNet-pi running but I need to be logged in to the pi via SSH.  If I close the terminal, the arecord and ffmpeg show as off and the application stops working.  Solution ?

**Reply by ehpersonal38** [reply]
This same issue was described in #375, here's the solution they arrived at:
`sudo systemctl enable getty@tty1.service`

**Reply by mcguirepr89** [reply]
Thanks @ehpersonal38 --

Out of curiosity, @PetrusMax, did you install using the [installation guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Installation-Guide) procedure?

**Reply by PetrusMax** [reply]
I did indeed and BirdNet was working well for the first few days.  It was only after an update (via tools)  that things went wrong.  Anyway,  after trying a few things,  I just reinstalled everything from the start and all is well again.
Next time I see an update, I might think twice about installing it ! 
thanks to all for the quick responses and support.

**Reply by mcguirepr89** [reply]
@PetrusMax Thank you for confirming your installation procedure. If you do attempt another update in the future via "Tools" > "System Controls" > "Update," I would like to ask you to save the output just in case there are issues during the update that can be identified through that output. I appreciate your patience with the software and hope you'll be able to participate in updates in the future once we can identify where things went awry. 

My very best regards,
Patrick

**Comment by PetrusMax** [comment]
Hi Patrick

Sure thing.  Lets stay in touch.  I would be happy to test for you. (Hope i can get a couple more pi’s to play with soon.)

Appreciate the nice work you are doing. 

Regards 
Peter 

sent from my ithing

**Comment by DCS17** [comment]
Hi,
I've got a very similar problem. Did update yesterday. Only "1" showing in the update page, but thought I'd stay up to date!
But now the spectrogram is empty and no recordings. Tried restarting core services and rebooting thru the Tools page. No joy.
Switched off at mains and restarted. No joy. 
arecord -l shows microphone as a device OK. 
Checked microphone working OK on my laptop computer and its is working and providing sound output.

Attached is output of the commands as outlined above. There is comment about bug in ALSA driver. Is that in any way relevant?
Grateful for any help. Don't really want to have to reinstall as got months worth of recordings stored.

Should say that initial install was from the #353 image, but all updates have been OK since install except this last one.
Thanks for any advice you may have.

[commands.txt](https://github.com/mcguirepr89/BirdNET-Pi/files/9377019/commands.txt)

**Reply by ehpersonal38** [reply]
It's interesting that the BirdNET server still seems to be analysing newly created .wav files and returning a confidence. Are the wav's just silent? Could you try playing one? They're located in `~/BirdSongs/August-2022/(today's date) `

Any ideas? @mcguirepr89

**Reply by DCS17** [reply]
Yes. The .wav files are silent when I listen to them. They are ~1.5MB in size.
The spectrogram is 'silent' too as are the "analysing now" spectrograms.
I've rechecked the mic and it's definitely working fine when plugged into laptop.
Thanks for your help.

**Reply by DCS17** [reply]
Update:
I've tried changing advanced settings Audio Card from default to the name produced by aplay -L.
Resulted in failure to start arecord service.
Tried a reboot - nothing.
Changed back to 'default' and reboot.
Success! I have sound and detections back :-)
No idea why that worked but I dont care :-)

**Reply by ehpersonal38** [reply]
Hmm, you know I do recall this happening a few times with a cheaper mic I had. It turned out the gain would sometimes drop to 0, so I had to go into alsamixer and raise it again. Guide here: https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card

So if it happens again do be sure to try that. Glad you got it working now though!

**Reply by DCS17** [reply]
Thanks for the pointer to alsamixer.
It's working OK now, so I'll quit messing ......... until the 'itch' gets the better of me :-)

**Comment by PetrusMax** [comment]
Hello there.    Does sound like a similar issues.  If you ssh into the pi and restart services from the command does the Birdnet thing start working again (when you remain logged in via ssh)?  If so, running the following command might help
sudo systemctl enable getty@tty1.service

I didnt try that as Id only recently built mine and didnt have so many recordings.  Hence it was easier to start over.  Anyway,  try it and let me know how it goes.  It will be useful for future reference and perhaps otehrs.

**Reply by DCS17** [reply]
Unfortunately no joy when ssh in and restarting services through terminal. Still a 'silent' spectrogram.
I have tried the getty@tty1.service and no change with that either :-(
I would think the microphone was dead, except that it works fine when attached to my laptop!
Thanks fro your suggestions though.

**Comment by PetrusMax** [comment]
Just noticed installation has failed again in a similar way to DCS17. Mic is working and server analysing sounds, giving a result and confidence level but not saving the recording.  Both a record and ffmpeg are off.  If I ssh into the pi and restart services it seems to work but fails the moment I exit terminal.  Need a permanent solution please.

**Reply by DCS17** [reply]
Hi,
Not sure that is the identical situation to me. The .wav files that are being analysed are silent in my case. Not had problem with ffmpeg services etc. It seems that the .wav generation is not reading properly from the mic.
From what you say it sounds as though the mic is working and producing sound to analyse (not silent?), but is failing to save the resulting analysis?

**Reply by ehpersonal38** [reply]
To confirm, you have executed the getty "enable" command on this new installation?

@mcguirepr89 I seek your help

**Reply by mcguirepr89** [reply]
Sure let's get some logs of the recording service and of the "--user" pulseaudio service and getty@tty1.service -- I'll follow up with that request as soon as I'm at a computer
